### ISSUE TITLE
Any way to resolve this composer conflict?

### ISSUE THREAD

**Issue by burzum** [lmc-eu/steward#134]
Steward requires `symfony/event-dispatcher:~3.0` but we're using Omnipay a pretty widely used payment lib as well and it requires guzzle/guzzle and guzzle again requires symfony/event-dispatcher 2.8.

I'm aware this is not really a problem of Steward but I don't know how to resolve this conflict. :(

Any advice? Is 3.0 really needed?

```sh
λ composer require lmc/steward
Using version ^2.1 for lmc/steward
./composer.json has been updated
Loading composer repositories with package information                                     Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install lmc/steward 2.1.0
    - Conclusion: remove symfony/event-dispatcher v2.8.20
    - Installation request for lmc/steward ^2.1 -> satisfiable by lmc/steward[2.1.0, 2.2.x-dev].
    - Conclusion: don't install symfony/event-dispatcher v2.8.20
    - lmc/steward 2.2.x-dev requires symfony/event-dispatcher ~3.0 -> satisfiable by symfony/event-dispatcher[3.0.x-dev, 3.1.x-dev, 3.2.x-dev, 3.3.x-dev, v3.0.0, v3.0.0-BETA1, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.0.8, v3.0.9, v3.1.0, v3.1.0-BETA1, v3.1.0-RC1, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.0-BETA1, v3.2.0-RC1, v3.2.0-RC2, v3.2.1, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.3.0-BETA1].
    - Can only install one of: symfony/event-dispatcher[3.0.x-dev, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[3.1.x-dev, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[3.2.x-dev, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[3.3.x-dev, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.0, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.0-BETA1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.2, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.3, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.4, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.5, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.6, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.7, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.8, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.0.9, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.0, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.0-BETA1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.0-RC1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.10, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.2, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.3, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.4, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.5, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.6, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.7, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.8, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.1.9, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.0, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.0-BETA1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.0-RC1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.0-RC2, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.1, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.2, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.3, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.4, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.5, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.6, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.7, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.2.8, v2.8.20].
    - Can only install one of: symfony/event-dispatcher[v3.3.0-BETA1, v2.8.20].
    - Installation request for symfony/event-dispatcher (locked at v2.8.20) -> satisfiable by symfony/event-dispatcher[v2.8.20].


Installation failed, reverting ./composer.json to its original content.
```

**Comment by burzum** [comment]
Nevermind, just created a new project / repo to implement the tests in. ;)

**Comment by OndraM** [comment]
@burzum Hi, I think there is a simple solution, and this is also a recommended practice: place Steward and your selenium tests into separate directory (in or inside your application repository) - eg. `selenium-tests/` and create new `composer.json` file here, which will contain only dependencies related to Steward and thus won't collide with dependencies of your app. If you don't need to reuse code from your app in you Selenium tests, this is the best solution.

**Comment by OndraM** [comment]
@burzum 👍 Right, this is what I just want to recommend you :).

**Comment by burzum** [comment]
Just a question, I've tried Nightwatch before but would prefer a php based Selenium solution. Nightwatch worked out of the box and was capable of running the tests without issues. It just opens the browser and runs the test as it should.

I've had the same trouble with Codeception and selenium before but don't remember how I solved it. The browser windows open but don't load the page, they stay on `data:,`.
